### DISCUSSION TITLE
Change value before validate

### DISCUSSION THREAD

**Discussion by bianchi** [react-hook-form/react-hook-form#3748] [Q&A] [answered]
I have a text field to read a decimal number. In Brazil comma is used as decimal separator. My field only allows user to input comma and numbers, that's working. Joi doesn't have an option to validate numbers separated by comma. My idea is to replace comma with dot before validate, and then, in case of errors, put the comma back. That seems a lot of trouble to do for each decimal input. Is there any other approach? If there isn't, where should I start?

**Comment by bluebill1049** [comment]
I think maybe you should consider using a Mask input component and wrapped with `Controller`.

there is an example in there as well: https://codesandbox.io/s/react-hook-form-v6-controller-qsd8r

**Reply by bianchi** [reply]
> I think maybe you should consider using a Mask input component and wrapped with `Controller`.
> 
> there is an example in there as well: https://codesandbox.io/s/react-hook-form-v6-controller-qsd8r

I think it is not the case. I do not have a pattern, user can input 15,7273838 or 1,5.

**Reply by bluebill1049** [reply]
so when are you going to replace the character? onChange or onBlur, I am a bit confused, it sounds like you want to manipulate user's input but then you are not.

**Reply by bianchi** [reply]
I can't replace It on blur or on change because user cannot see that I changed comma for dot. I only need to change the value to validate, since i cannot validate numbers with comma. Options I see is validate with regex but i Will lose min, Max validations.

**Reply by bluebill1049** [reply]
hmmm so can you format the value before the validation at resolver? maybe you can build a custom one?

**Reply by bianchi** [reply]
> hmmm so can you format the value before the validation at resolver? maybe you can build a custom one?

Never tried that but it seems the way to go. Will give a try. Thanks for your help and keep up the good work

**Reply by bluebill1049** [reply]
no worries, let me know if that's working out for you or not.

**Comment by bianchi** [comment, marked as answer]
Custom resolver worked. Ty @bluebill1049 
```
const {handleSubmit, errors, reset, control, watch} = useForm<CamposForm>({
		resolver: async data => {
			data.estoque_minimo = data.estoque_minimo.replace(',', '.');
			data.estoque_maximo = data.estoque_maximo.replace(',', '.');
			data.estoque_ponto_pedido = data.estoque_ponto_pedido.replace(',', '.');

			const {error, value: values} = validacao.validate(data, {
				abortEarly: false,
			});

			return {
				values: error ? {} : values,
				errors: error
					? error.details.reduce((previous, currentError) => {
							return {
								...previous,
								[currentError.path[0]]: currentError,
							};
					  }, {})
					: {},
			};
		},
		criteriaMode: 'all',
		shouldUnregister: false,
	});
```

**Reply by bluebill1049** [reply]
awesoem to hear.

**Reply by AlexandreGuidin** [reply]
Same country, same problem.

I dont know why Chrome can validate number fields with `.` or `,` but safari doest not.
Trying to understand the differences.

**Reply by stayhungry07212** [reply]
@bianchi , can you please share a more code? How about a gist? Thanks in advance.
